src/book/03-guides/04-general/03-glossary.mdx

```diff
@@ -5,3 +5,181 @@ title: Glossary
+
+[**Transformation**](<https://en.wikipedia.org/wiki/Transformation_(function)>): A function which maps a set to itself (sometimes called an operator on that set). Often, it is used in the context of a spatial manipulation - regularly representated as a translation and a rotation.
+
+[**Zero Moment Point (ZMP)**](https://en.wikipedia.org/wiki/Zero_moment_point): With relation to walking motion, it is the point with respect to which the contact force of the foot on the ground produces zero moment in the horizontal direction.
+
+### External
+
+[**Arch**](https://www.archlinux.org/): A lightweight linux distribution which the NUgus robots use as their primary operating system.
+
+[**Bit-Bots**](https://bit-bots.de/): A RoboCup team from the University of Hamburg, which created the quintic walk NUbots uses.
+
+[**Armadillo (arma)**](http://arma.sourceforge.net/): The C++ linear algebra library developed by the CSIRO, previously used by NUbots. Documentation can be found [here](http://arma.sourceforge.net/docs.html).
+
+[**NimbRo**](http://www.nimbro.net/): A RoboCup team from the University of Bonn, Germany.
+
+[**Buildkite**](https://buildkite.com/): A platform for running fast, secure, and scalable continuous integration pipelines, which NUbots runs on machines in the NUbots lab.
+
+[**Clang**](https://clang.llvm.org/): A compiler, frontend, and tooling infrastructure used for C and C++ which is part of the LLVM infrastructure.
+
+[**ClangFormat**](https://clang.llvm.org/docs/ClangFormat.html): A software which formats source code according to a given set of rules.
+
+[**CM740**](<http://support.robotis.com/en/product/robotis-op2/sub_controller(cm-740).htm>): A _Robotis_ sub-controller which gives the commands to the servos.
+
+[**CMake**](https://cmake.org/): An open-source build system generator NUbots uses to configure and build the main codebase.
+
+[**Darwin**](https://robots.ieee.org/robots/darwin/): A _Robotis_ robot platform which was the platform NUbots used before making the NUgus.
+
+[**Docker**](https://www.docker.com/): A software service for developing containers which enable standardised cross-platform development consistency.
+
+[**Dockerhub**](https://hub.docker.com/): A hosting service for docker container images.
+
+[**Dynamixel**](https://www.robotis.us/dynamixel/): The _Robotis_ servo models which NUgus uses.
+
+[**igus**](https://www.igus.com.au/): The creators of the robotics platform NUgus is based on.
+
+[**igus Humanoid Open Platform**](https://arxiv.org/abs/1809.11110): The 3d printed base design which the NUgus is based on.
+
+[**Eigen**](https://eigen.tuxfamily.org/): The C++ linear algebra library NUbots uses. Documentation can be found [here](https://eigen.tuxfamily.org/dox/index.html).
+
+[**Gazebo**](https://gazebosim.org/): A simulation software used which was used to optimise the "OldWalkEngine" with a genetic algorithm.
```

Comment:
```suggestion
[**Gazebo**](https://gazebosim.org/): A simulation software that is used to optimise motions with a genetic algorithm.
```

Reply:
I don't agree with this one. This makes it seem like that's the point of gazebo, but gazebo is a lot more than just that project - especially this year with RoboCup